src/Nethermind/Nethermind.Core2/Containers/BeaconState.cs

```diff
@@ -0,0 +1,122 @@
+//  Copyright (c) 2018 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using Nethermind.Core.Extensions;
+using Nethermind.Core2.Crypto;
+using Nethermind.Core2.Types;
+
+namespace Nethermind.Core2.Containers
+{
+    public class BeaconState
+    {
+        public static int SszDynamicOffset = sizeof(ulong) +
+                                            Slot.SszLength +
+                                            Fork.SszLength +
+                                            BeaconBlockHeader.SszLength +
+                                            2 * Time.SlotsPerHistoricalRoot * Sha256.Size +
+                                            sizeof(uint) +
+                                            Eth1Data.SszLength +
+                                            sizeof(uint) +
+                                            sizeof(ulong) +
+                                            2 * sizeof(uint) +
+                                            Time.EpochsPerHistoricalVector * Sha256.SszLength +
+                                            Time.EpochsPerSlashingsVector * Gwei.SszLength +
+                                            2 * sizeof(uint) +
+                                            1 + // not sure
+                                            3 * Checkpoint.SszLength;
+
+        public static int SszLength(BeaconState container)
```

Comment:
Why are those static when they are instance based? Maybe better to have some interface with SszLength property on those classes?

Reply:
good question...

Reply:
aaah, I know why - so we can use const in most of the cases instead of the instance property

Reply:
not sure if the compiler would optimize away const properties...